JsonSchema.Generation.Tests/TestEnvironment.cs

```diff
@@ -1,24 +1,23 @@
-﻿using System.Text.Json;
+﻿using System.Text.Encodings.Web;
+using System.Text.Json;
 using System.Text.Json.Serialization;
 
-namespace Json.Schema.Generation.Tests
-{
-	internal static class TestEnvironment
-	{
-		public static readonly JsonSerializerOptions SerializerOptions = new()
-		{
-			TypeInfoResolverChain = { GenerationTestsJsonSerializerContext.Default, JsonSchema.TypeInfoResolver },
-		};
+namespace Json.Schema.Generation.Tests;
 
-		public static readonly JsonSerializerOptions SerializerOptionsWriteIndented = new(SerializerOptions)
+internal static class TestEnvironment
+{
+	public static readonly JsonSerializerOptions SerializerOptions =
+		new()
 		{
-			WriteIndented = true
+			TypeInfoResolverChain = { TestSerializerContext.Default },
+			WriteIndented = true,
+			Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping
 		};
-	}
-
-	[JsonSerializable(typeof(IJsonSchemaKeyword))]
-	internal partial class  GenerationTestsJsonSerializerContext : JsonSerializerContext
-	{
-		
-	}
 }
+
+[JsonSerializable(typeof(JsonSchema))]
+[JsonSerializable(typeof(PropertiesKeyword))]
+[JsonSerializable(typeof(int))]
+[JsonSerializable(typeof(string))]
+[JsonSerializable(typeof(bool))]
```

Comment:
Not sure why I had to add these.  They're all used only internally to serializing a `JsonNode`.  Adding `JsonNode` here didn't help.

Reply:
@eiriktsarpalis ? I suspect that JsonNode's converter looks up TypeInfos in the options dictionary even for these POD types. @gregsdennis if you can share a full call stack (turning off just my code) for when we fail without these, that would be helpful for Eirik to take a look at.

Reply:
Yeah, it's one of the unfortunate issues with `JsonNode` design. It has been designed to encapsulate arbitrary .NET values so depending on how the `JsonNode` is constructed it can be representing JSON numbers or strings as `JsonValue<int>` and `JsonValue<string>`. From that point on, there's no way it can serialize itself without being able to obtain a `JsonTypeInfo<T>` from the underlying `JsonSerializerOptions`.

If this has been caused by something like
```C#
new JsonArray { 42, "string", false }
```
you could try to replace it either with
```C#
new JsonArray { (JsonNode)42, (JsonNode)"string", (JsonNode)false };
```
or just
```C#
JsonNode.Parse("""[42, "string", false]""")
```

Reply:
Oh, yeah, we changed a bunch of array initializers with ctor calls before. I'll have a scan for any that we missed.

Reply:
(That kinda defeats the purpose of the implicit cast, though, doesn't it?)

Reply:
I debugged this a little, the problem is a little subtle. It's that the JsonSchema.Generation library is calling JsonNode.Create(object) and then JsonSchema tries to serialize that later. Because the options came from a JsonSerializerContext, we only look for those POD types out of that options dictionary. We could try to call a helper instead of JsonValue.Create everywhere in our code, like this (below). I tried this locally and it seems to address the problem and lets me remove the primitive types from the client JsonSerializerContext.
```
	public static JsonValue? Create(object? value)
	{
		if (value is bool b)
		{
			return JsonValue.Create(b);
		}
		else if (value is double d)
		{
			return JsonValue.Create(d);
		}
		else if (value is int i)
		{
			return JsonValue.Create(i);
		}
		else if (value is string s)
		{
			return JsonValue.Create(s);
		}
		else
		{
			return JsonValue.Create(value);
		}
	}
```

Reply:
(And if you go this path, we should add in the missing POD types that JsonValue supports as builtin. (this list is missing char and float, etc)

Reply:
Looks like that call is just in the `ConstIntent` and `DefaultIntent`, which actually comes from the associated attributes.

I'd like to:

- change the intent class ctors to take `JsonNode` instead, since that's what the `ConstKeyword` takes
- create strongly typed overloads for the attribute constructors that then immediately store the value into a `JsonNode` that's then passed into the intent.

Reply:
I saw that some of the object-based JsonNodes were coming through `public static JsonSchemaBuilder Const(this JsonSchemaBuilder builder, JsonNode? node)` too.